cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/SourceControlMetadataStore.java

```diff
@@ -142,6 +175,18 @@ public void deleteAll(String namespace) throws IOException {
     getNamespaceSourceControlMetadataTable().deleteAll(getNamespaceRange(namespace));
   }
 
+  /**
+   * Deletes the source control metadata associated with the specified repository metadata ID.
+   *
+   * @param repoMetadataId The {@link RepoMetadataId} for which source control metadata needs
+   *                       to be deleted.
+   * @throws IOException If an I/O error occurs while deleting the source control metadata.
+   */
+  public void deleteRepoSourceControlMeta(RepoMetadataId repoMetadataId) throws IOException {
+    getRepositorySourceControlMetadataTable().delete(
+        getRepoSourceControlMetadataPrimaryKey(repoMetadataId));
+  }
+
   private Collection<Field<?>> getNamespaceSourceControlMetaFields(ApplicationId appId,
```

Comment:
Use `ApplicationReference` than `ApplicationId` as `ApplicationReference` identifies versionless application.

Reply:
Right. Thank you for this correction. Fixed